src/cmd/manage_nsfs.js

```diff
@@ -257,15 +257,23 @@ async function delete_bucket(data, force) {
     try {
         const temp_dir_name = native_fs_utils.get_bucket_tmpdir_name(data._id);
         const bucket_temp_dir_path = native_fs_utils.get_bucket_tmpdir_full_path(data.path, data._id);
-        const entries = await nb_native().fs.readdir(fs_context_fs_backend, data.path);
-        const object_entries = entries.filter(element => !element.name.endsWith(temp_dir_name));
-        if (object_entries.length === 0 || force) {
-            await native_fs_utils.folder_delete(bucket_temp_dir_path, fs_context_fs_backend, true);
-            await native_fs_utils.delete_config_file(fs_context_config_root_backend, buckets_dir_path, bucket_config_path);
-            write_stdout_response(ManageCLIResponse.BucketDeleted, '', { bucket: data.name });
-            return;
+        let entries;
+        try {
+            entries = await nb_native().fs.readdir(fs_context_fs_backend, data.path);
+        } catch (err) {
+            dbg.warn(`delete_bucket: bucket name ${data.name}, path: ${data.path} got an error:`, err);
+            // if the bucket's path was deleted first (encounter ENOENT error) - continue deletion
+            if (err.code !== 'ENOENT') throw err;
```

Comment:
I will add more details in the `dbg.warn` message.

The flow is the same as it was before, if the error was thrown in this line `await nb_native().fs.readdir`, it will be caught and thrown again if it doesn't `ENOENT` error, and then it will be caught by the catch of the function and thrown again if it doesn't `ENOENT`, then it would be mapped `FS_ERRORS_TO_MANAGE` in case it is not there it would be `InternalError`.